sdk/servicebus/service-bus/review/service-bus.api.md

```diff
@@ -29,6 +31,26 @@ export type AuthorizationRule = {
     secondaryKey?: string;
 };
 
+// @public
+export interface Closeable {
+    close(): Promise<void>;
+}
+
+// @public
+export type ContextType<LockModeT> = LockModeT extends "peekLock" ? ContextWithSettlement : LockModeT extends "receiveAndDelete" ? {} : never;
+
+// @public
+export interface ContextWithSettlement {
+    abandon(message: ReceivedMessage, propertiesToModify?: {
+        [key: string]: any;
+    }): Promise<void>;
+    complete(message: ReceivedMessage): Promise<void>;
+    deadLetter(message: ReceivedMessage, options?: DeadLetterOptions): Promise<void>;
```

Comment:
In .Net we have one more overload for deadLetter where we take `message` and `propertiesToModify` as parameters. Do we have `propertiesToModify` in `DeadLetterOptions`?

Reply:
I think is just an oversight, I'll check.

deadLetter(message, options =  {
  propsToModify,
})

Reply:
Feature team - RECONCILE.

Reply:
JS
![image](https://user-images.githubusercontent.com/10452642/76128278-b98f0300-5fb8-11ea-8c92-aa78b69783e3.png)

Python 
```python
def dead_letter(self, description : str = None)
```

Reply:
.Net
```csharp

public virtual Task DeadLetterAsync(ServiceBusReceivedMessage message, IDictionary<string, object> propertiesToModify = null, CancellationToken cancellationToken = default);

public virtual Task DeadLetterAsync(ServiceBusReceivedMessage message, string deadLetterReason, string deadLetterErrorDescription = null, CancellationToken cancellationToken = default);
```

Reply:
After talking about this on our team we've decided to incorporate all the fields into a single object - this avoids making the user pass two object literals in a row (which isn't great) and also takes advantage of TS interfaces. :)

So this'll be the final signature:

deadLetter(options?: { [k: string]: any } & DeadLetterOptions)

So the user can do this:

deadLetter({
  deadLetterReason: "hello",
  customProperty: "hello again!"
})